### ISSUE TITLE
aiProcess_GenSmoothNormals is very slow in case of huge meshes

### ISSUE THREAD

**Issue by jjakubovic** [assimp/assimp#4523]
When aiProcess_GenSmoothNormals flag is set during import of huge PLY mesh (285 000 000 tris, 5 GB), it takes too long to import it: 1h30m. Without it the import takes 10m. In case of OBJ version of this mesh, when also aiProcess_JoinIdenticalVertices flag was set, the import didn’t even finish after 24h (it was killed). Without these flags the OBJ import took 15m.

I can’t share the mesh, it’s from a customer, sorry, but it was a mesh from photogrametry with vertex colors.

Used import flags:
aiProcess_ValidateDataStructure
aiProcess_Triangulate
aiProcess_SortByPType
aiProcess_FlipUVs
aiProcess_GenSmoothNormals 

Assimp version: 5.2.3.

**Comment by kimkulling** [comment]
Do you have any example model?

**Comment by jjakubovic** [comment]
I don't have the rights to this model, so I can't share it, but I'll ask the artist whether he has different model with similar parameters that could be shared.

**Comment by jjakubovic** [comment]
I've managed to get similar model from artist that can be shared. Here it is: https://drive.google.com/file/d/1Pv_U07201qLD1RKqI6_cHJsUe8mkvN-o/view?usp=sharing

**Comment by jjakubovic** [comment]
I've managed to get similar model from artist that can be shared. Here it is: https://drive.google.com/file/d/1Pv_U07201qLD1RKqI6_cHJsUe8mkvN-o/view?usp=sharing. Sharing is restricted, so please ask me using that link if you need access.

**Comment by tellypresence** [comment]
@jjakubovic Is this still a problem?

**Comment by jjakubovic** [comment]
Yes, I've tested it on latest Assimp (5.4.3) and it still happens. Without aiProcess_GenSmoothNormals the file linked above loads (Importer::ReadFile) in 31 min. With aiProcess_GenSmoothNormals the load was still running after 6 hours and then I killed it.

**Comment by tellypresence** [comment]
PLY issues tracked via epic #6163
